ouroboros-consensus-cardano/README.md

```diff
@@ -7,9 +7,226 @@ particular for the Byron, Shelley and Cardano instantiations.
+
+### Running the tool
+
+When you run db-analyser without any arguments, it will print out a help
+message. The tool requires a few command line options. Some of these options
+determine which analysis to perform on the ChainDB. In the following sections,
+we explain each command line option.
+
+#### --db PATH
+
+The tool works on a cardano-node's ChainDB. Thus the user must provide an obligatory `--db PATH` argument pointing to the particular DB.
+
+#### --verbose
+
+db-analyser will get quite noisy
+
+#### --only-immutable-db
+
+By default db-analyser will process all blocks from the chain database
+(`ChainDB`), from the genesis block up to the chain tip. In order to do this it
+must first properly initialize the whole database. That means that before it
+even starts processing blocks it will:
+
+1. look for latest snapshot stored in DB_PATH/ledger
+2. load that snapshot into memory
+3. start replaying blocks
+   * starting from that ledger state
+   * while updating the ledger state in the process for each replayed block
+   * keeping the intermediate results (ledger states) in memory while replaying blocks that live in the volatile DB (less than k blocks from the tip)
+
+This may heavily impact any profiling that the user might be interested in doing.
+
+To counter that problem `--only-immutable-db` flag was introduced.
+
+```
+[--only-immutable-db [--analyse-from SLOT_NUMBER]]
+```
+
+When enabled, db-analyser will work only with blocks from immutableDB, thus initialization described above will not happen.
+
+This flag comes with an additional `--analyse-from` flag. It allows to start processing blocks from the requested slot number. A snapshot at that slot number must exist in `DB_PATH/ledger/SLOT_NUMBER_db-analyser` - where `SLOT_NUMBER` is the value provided by the user with the `--analyse-from` flag.
+The user can use snapshots created by the node or they can create their own snapshots via db-analyser - see the `--store-ledger` command
+
+#### COMMAND
+
+There are three options: `byron`, `shelley`, `cardano`. When in doubt which one to use, use `cardano`.
+
+* `byron`
+
+User should run this if they are dealing with Byron only chain. When the command is `byron` then user must provide `--configByron PATH` pointing to a byron configuration file.
+
+* `shelley`
+
+User should run this if they are dealing with Shelley only chain (neither Byron nor Allegra or any other era that comes after). When the command is `shelley` then user must provide `--configShelley PATH` pointing to a shelley configuration file. They may also provide `--genesisHash HASH` and `--threshold THRESHOLD`
+
+* `cardano`
+User should run this if they are dealing with a `cardano` chain.
+
+#### --num-blocks-to-process
+
+```
+[--num-blocks-to-process INT]
+```
+
+The user can limit the maximum number of blocks that db-analyser will process.
+
+#### Analysis
+
+Lastly the user must provide the analysis that should be run on the chain:
```

Comment:
Good point. If no analysis flag is provided then the `OnlyValidation` analysis is run. I'll document that.

Reply:
It would also be nice to document what `OnlyValidation` means and how it is related to `validate-all-blocks` and `minimum-block-validation`

I think those flags together caused the most confusion for me when I was trying to understand what db-analyser does

Reply:
Damian included a section, but there also is #35 which is tracking this and other improvements to validation- and ledger-related db-analyser docs.